### ISSUE TITLE
Investigate possible Null Ref when crafting items from `Dead Man's Switch`

### ISSUE THREAD

**Issue by Sn1p3rr3c0n** [zymex22/Project-RimFactory-Revived#823]
> When I use the the assemblers to craft weapons from the Dead Man's Switch (DMS) mod, I get an error that causes the progress bar to hang at 100%. Is it due to the weapons from DMS not having a quality value?

Reported by `ShinUon` on Steam. Needs Confirmation and actual log.
https://steamcommunity.com/workshop/filedetails/?id=3121742525
```
System.NullReferenceException: Object reference not set to an instance of an object
[Ref E9D48B8A]
at ProjectRimFactory.AutoMachineTool.GenRecipe2.PostProcessProduct (Verse.Thing product, Verse.RecipeDef recipeDef, ProjectRimFactory.AutoMachineTool.IRecipeProductWorker worker, RimWorld.Precept_ThingStyle precept) [0x00077] in <e85aa4e119764ab3ac1cf233e1b3a454>:0
at ProjectRimFactory.AutoMachineTool.GenRecipe2+<MakeRecipeProductsInt>d__1.MoveNext () [0x002fc] in <e85aa4e119764ab3ac1cf233e1b3a454>:0
at <0x1c551fbeaf0 + 0x002c9> <unknown method>
```

**Comment by Sn1p3rr3c0n** [comment]
> If this has already been reported. with Science Never Stops True Pinnacle of creation. the SAL autocrafters fail to output on the Utopia Device. however I have isolated it to two recipes. the TERMINUS and the BEHOLDER.
I have reduced my mod list to only TPOC, rimfactory, and precursor mods for those to work.
Reported by `Maelchlor` on Steam

To me it seems this might be the same issue

**Comment by Sn1p3rr3c0n** [comment]
resolved by #824
